uaclient/files/notices.py

```diff
@@ -0,0 +1,206 @@
+        """
+        if self.is_root:
+            label = notice_details.label
+            file_order_id = notice_details.order_id
+            is_notice_permanent = notice_details.is_permanent
+            if not self.directory:
+                self.directory = (
+                    defaults.NOTICES_PERMANENT_DIRECTORY
+                    if is_notice_permanent
+                    else defaults.NOTICES_TEMPORARY_DIRECTORY
+                )
+            notice_files = self.get_all_notice_files()
+            if notice_files is None:
+                os.makedirs(self.directory)
+                notice_files = []
+            labels_present = [file.split("-")[-1] for file in notice_files]
+            if label in labels_present:
+                return
+            filename = "{}-{}".format(file_order_id, label)
+            system.write_file(
+                os.path.join(self.directory, filename), description
+            )
+        else:
+            event.warning("Trying to add a notice as non-root user")
+            raise exceptions.NonRootUserError
+
+    def remove(self, notice_details: NoticeFileDetails):
+        """Deletes a notice file.
+
+        :param notice_details: Holds details concerning the notice file.
+        :param description: The content to be written to the notice file.
+        :raises: NonRootUserError when called by a user who is not root.
+        """
+        if self.is_root:
+            label = notice_details.label
+            file_order_id = notice_details.order_id
+            is_notice_permanent = notice_details.is_permanent
+            if not self.directory:
+                self.directory = (
+                    defaults.NOTICES_PERMANENT_DIRECTORY
+                    if is_notice_permanent
+                    else defaults.NOTICES_TEMPORARY_DIRECTORY
+                )
+            filename = "{}-{}".format(file_order_id, label)
+            notice_files = self.get_all_notice_files()
+            if notice_files:
+                label_file = [
+                    file for file in notice_files if file == filename
+                ]
+                if len(label_file) == 1:
+                    system.remove_file(
+                        os.path.join(self.directory, label_file[0])
+                    )
+        else:
+            event.warning("Trying to remove notice as non-root user")
+            raise exceptions.NonRootUserError
+
+    def get_all_notice_files(
+        self, notice_dir: Optional[str] = None
+    ) -> Optional[List[str]]:
+        notice_dir = notice_dir if notice_dir else self.directory
+        if not os.path.exists(notice_dir):  # type: ignore
+            return None
+
+        notices = [
+            file
+            for file in os.listdir(notice_dir)
+            if os.path.isfile(os.path.join(notice_dir, file))  # type: ignore
+        ]
+        notices.sort()
+        return notices
+
+    def read(self):
```

Comment:
I'd like this to have a typed return value. And I'd like the return type to include the id and ordering number of each notice in addition to the notice message contents